From 36edf0678a7d3d8490fb44059221d4de10f40ed4 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Mon, 16 Jan 2017 00:34:33 +0000
Subject: [PATCH 2/2] doc: Add restrictions for using or reconfiguring apt

nss-mdns should use this in its autopkgtests, which exercise
different orderings for installation of libnss-mdns vs. libnss-resolve,
and make sure that upgrading from jessie's libnss-mdns works.
---
 doc/README.package-tests.rst | 15 +++++++++++++++
 runner/autopkgtest           |  4 +++-
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/doc/README.package-tests.rst b/doc/README.package-tests.rst
index 3117f3d..bac0a91 100644
--- a/doc/README.package-tests.rst
+++ b/doc/README.package-tests.rst
@@ -217,6 +217,21 @@ uses-network
     some cases a test attempts to contact the network as a side-effect,
     for example resolving a hostname in DNS.
 
+needs-apt
+    The test needs to use apt to install or remove packages during
+    testing. The configured ``Depends`` will be installed initially,
+    but other packages might be added or removed during testing.
+    The test should also declare ``needs-root``.
+
+reconfigures-apt
+    The test reconfigures the apt sources, for example to enable an
+    older suite and test an upgrade. This restriction implies
+    ``needs-apt``. The test should also declare ``needs-root``.
+    A test with this restriction may assume that the configured
+    apt mirror has the other suites that would normally be available
+    from the same distro: for example, a test running in Debian
+    unstable might rely on the same mirror having Debian stable.
+
 Defined features
 ----------------
 
diff --git a/runner/autopkgtest b/runner/autopkgtest
index 4bed5cb..12edc1d 100755
--- a/runner/autopkgtest
+++ b/runner/autopkgtest
@@ -163,7 +163,9 @@ def run_tests(tests, tree):
                          opts.build_parallel)
         if not t.result:
             errorcode |= 4
-        if 'breaks-testbed' in t.restrictions:
+        if ('breaks-testbed' in t.restrictions or
+            'needs-apt' in t.restrictions or
+            'reconfigures-apt' in t.restrictions):
             testbed.needs_reset()
 
     testbed.needs_reset()
-- 
2.11.0

